packaging/common/rabbitmq-server-ha.ocf

```diff
@@ -0,0 +1,1585 @@
+
+    # remove old PID-file if it exists
+    if [[ -f $OCF_RESKEY_pid_file ]] ; then
+        ocf_log warn "${LH} found old PID-file '${OCF_RESKEY_pid_file}'."
+        pid=$(cat ${OCF_RESKEY_pid_file})
+        if [[ -d /proc/${pid} && ! -z ${pid} ]] ; then
+            ocf_run cat /proc/${pid}/cmdline | grep -c 'bin/beam' 2>&1 > /dev/null
+            rc=$?
+            if [[ $rc == $OCF_SUCCESS ]] ; then
+                ocf_log warn "${LH} found beam process with PID=${pid}, killing...'."
+                ocf_run kill -9 $pid
+            else
+                ocf_log err "${LH} found unknown process with PID=${pid} from '${OCF_RESKEY_pid_file}'."
+                return $OCF_ERR_GENERIC
+            fi
+        fi
+        ocf_run rm -rf $OCF_RESKEY_pid_file
+    fi
+
+    [ -f /etc/default/rabbitmq-server ] && . /etc/default/rabbitmq-server
+
+    # run beam process
+    local command="${OCF_RESKEY_binary} >> \"${OCF_RESKEY_log_dir}/startup_log\" 2>/dev/null"
+    RABBITMQ_NODE_ONLY=1 su rabbitmq -s /bin/sh -c "${command}"&
+    ts_end=$(( $(now) + ${OCF_RESKEY_start_time} ))
+    rc=$OCF_ERR_GENERIC
+    while [ $(now) -lt ${ts_end} ]; do
+        # waiting for normal start of beam
+        pid=0
+        pf_end=$(( $(now) + 3 ))
+        while [ $(now) -lt ${pf_end} ]; do
+            # waiting for OCF_RESKEY_pid_file of beam process
+            if [[ -f $OCF_RESKEY_pid_file ]] ; then
+                pid=$(cat ${OCF_RESKEY_pid_file})
+                break
+            fi
+            sleep 1
+        done
+        if [[ $pid != 0 && -d /proc/${pid} ]] ; then
+            rc=$OCF_SUCCESS
+            break
+        fi
+        sleep 2
+    done
+    if [[ $rc != $OCF_SUCCESS ]]; then
+        if [[ "${pid}" == "0" ]] ; then
+            ocf_log warn "${LH} PID-file '${OCF_RESKEY_pid_file}' not found"
+        fi
+        ocf_log err "${LH} RMQ-runtime (beam) didn't start succesfully (rc=${rc})."
+    fi
+
+    return $rc
+}
+
+check_plugins() {
+  # Check if it's safe to load plugins and if we need to do so. Logic is:
+  #   if (EnabledPlugins > 0) and (ActivePlugins == 0) ; then it's safe to load
+  # If we have at least one active plugin, then it's not safe to re-load them
+  # because plugins:setup() would remove existing dependency plugins in plugins_expand_dir.
+  ${OCF_RESKEY_ctl} eval '{ok, EnabledFile} = application:get_env(rabbit, enabled_plugins_file), EnabledPlugins = rabbit_plugins:read_enabled(EnabledFile), ActivePlugins = rabbit_plugins:active(), if length(EnabledPlugins)>0 -> if length(ActivePlugins)==0 -> erlang:error("need_to_load_plugins"); true -> false end; true -> false end.'
+  return $?
+}
+
+load_plugins() {
+  check_plugins
+  if [[ $? == 0 ]] ; then
+    return 0
+  else
+    ${OCF_RESKEY_ctl} eval 'ToBeLoaded = rabbit_plugins:setup(), ok = app_utils:load_applications(ToBeLoaded), StartupApps = app_utils:app_dependency_order(ToBeLoaded,false), app_utils:start_applications(StartupApps).'
```

Comment:
Can't you use `rabbitmq-plugins` for this?

Should we add more options and output formats to the command?